accord-core/src/main/java/accord/impl/InMemoryCommandStore.java

```diff
@@ -461,6 +461,19 @@ void update(Ranges add)
         }
     }
 
+    public AsyncChain<Timestamp> maxAppliedFor(Seekables<?, ?> keysOrRanges, Ranges slice)
+    {
+        Seekables<?, ?> sliced = keysOrRanges.slice(slice, Minimal);
+        Timestamp timestamp = Timestamp.NONE;
+        for (GlobalCommand globalCommand : commands.values())
```

Comment:
I believe that's intentional, since we're only interested in commands that have been applied to disk and range txns are (currently) only used as markers

Reply:
> range txns are (currently) only used as markers/sync points/etc

The only caller atm is `org.apache.cassandra.service.accord.AccordFetchCoordinator` (outside of tests), which is focused on streaming user data... so unless we have range write support (think we only plan for range read?), then the only data present should be from key commands... so excluding range commands should be fine in this context...

So, assuming that we do only care about what data was applied to user tables in this context, don't we also want to filter out non-write txn?  `org.apache.cassandra.service.accord.AccordCommandStore#maxApplied` does, but not `accord.impl.InMemoryCommandStore#maxAppliedFor`?

Reply:
double checked with Ariel and at least for v1 we don't plan to have user level range reads/writes and his work is a metadata txn, so excluding range txn should still be fine after his work

Reply:
I think this is actually potentially a bug. Range transactions take the place of key transactions on aggregate, including for the purpose of bootstrap. Let's say you bootstrap from node A to node B, then from node B to node C, and let's say you don't receive any new transactions for some range in between. You'll have a maxApplied of NONE, rather than the range transaction that brought data from A to B.

I'm not sure if this would in practice lead to any isolation violations, but we probably do want to include range transactions as they (pretty much only) provide ordering guarantees with respect to other transactions right now.

Reply:
Hmm, thinking about this we probably want to propagate the maxApplied we received to any future owners, rather than the range transactions.

Reply:
I got mixed up here. `InMemoryCommandStore.commands` contains _all_ commands, key and range alike. CommandsForKey/RangeCommands are where they're separate, so this should be ok

Reply:
I was mistaken too though, I don’t think this is sufficient to propagate maxApplied reliably, as the range transaction that is used for bootstrap may be less than a propagated command. We probably want an additional register to store a timestamp to propagate. Perhaps I’ll fold that into my truncation patch as we also need one for tracking timestamps that have been truncated.

Reply:
Hmm, actually no. In this case we would not be "safe to read". So we're fine. Forget I said anything.